### PR TITLE
Rename package to Proj.jl

### PR DESCRIPTION
**Pull request by visr**
This renames the package to Proj.jl as discussed in #26. Proj4.jl currently wraps both the old and new API, and uses PROJ version 7.2. This also updates to PROJ 9.0. Since PROJ 8 the old API is removed however, making this a breaking release, and in my opinion a good time for the rename. This also will set the minimum supported Julia version to 1.6, since the PROJ_jll has that as well.

The new (Transformation / CoordinateTransformations) API is kept the same here. The wrapper generation code has been updated however, using the new Clang.jl. All the generated code is together in libproj.jl. This includes the geodesic code, all the geod_ * functions are there. I have not yet generated higher level geodesic code like in [proj_geodesic.jl](https://github.com/JuliaGeo/Proj4.jl/blob/9a63d757cf77a839bee336f398efd21928a4975f/src/proj_geodesic.jl), perhaps it's fine to leave that for the future? If there is anything else important missing here, it'd be good to hear about that here.

This passes tests now, but I still marked it as a draft to play around a bit more with the generated code. Currently the only real replacement we do is to replace `PJ_COORD` with `Coord`, which is defined as `const Coord = SVector{4, Float64}`. But perhaps we should give such a treatment also to other types like `PJ` and `PJ_AREA`, which are now a `Ptr{Cvoid}`, or `PJ_PROJ_INFO`. The PROJ data types are documented [here](https://proj.org/development/reference/datatypes.html). Input very welcome. Some of this behavior can also be controlled by [Clang.jl generator options](https://github.com/JuliaInterop/Clang.jl/blob/331313d58c39d102aa0bc7b09a9d1473f16d373e/gen/generator.toml#L153-L157).

Another TODO is to integrate `build_docstring` from wrap_proj.jl and then delete this file in favor of generator.jl. Currently we use Clang.jl's new feature to parse the doxygen docstrings and add them, but we can probably improve them by parsing the Doxygen XML, this combination is also done in GDAL.jl. Although if it's not much improved I'd be happy to get rid of that code as well. Or maybe just write some good docstrings manually for the most important functions, and have generator.jl add them to libproj.jl. *Edit*: it does add quite a few more docstrings with the old doxygen.xml code, but I decided not to add it. Some docstrings were malformed. Plus I think that this code makes maintenance of this package harder.

### CODE REVIEW

**Review by c42f — COMMENTED** [review]
Very cool. The generated docs look pretty useful already.

> which is defined as `const Coord = SVector{4, Float64}`

By the way I did this recently https://github.com/JuliaArrays/StaticArrays.jl/pull/980 ... which for Proj could be seen as useful (for projected CRSs) or confusing (for everything else).  You're welcome / I'm sorry respectively :laughing:

File: src/Proj.jl [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+using PROJ_jll
+using CEnum
+using StaticArrays
+using CoordinateTransformations
+
+export PROJ_jll
+
+# re-export CoordinateTransformations methods we implement
+export compose, ∘
```
[the remark this answers is not in the document]
Comment (visr) [reply src/Proj.jl]: Thanks, was unaware that `∘` is exported from Base. I removed both exports in 094c6d08115da64c99238a4cad5206d68934547d.

**Comment by visr** [comment]
> By the way I did the following recently in StaticArrays: JuliaArrays/StaticArrays.jl#980 ... which for Proj could be seen as useful (for projected CRSs) or confusing (for everything else). You're welcome / I'm sorry respectively 😅

Ha yes I saw that! I think for many cases it's quite nice to use. Perhaps we should document this, but then also warn about axis order. And `w` stands for `when` to give the time coordinate ;)

**Comment by c42f** [comment]
> And w stands for when to give the time coordinate

Haha :-) I did joke that we were not going to use `x,y,z,t` in that issue... but I was thinking that almost nobody would be wanting spacetime coordinates because few people would be doing special relativity with StaticArrays. Somehow I forgot about geodesy.

**Comment by asinghvi17** [comment]
I was working on GeoMakie, and noticed that there is one feature which doesn't work all the time: inverse transformations. With some transformations, Proj errors out when applying the inverse transformation - the error message is something along the lines of "cannot construct a forward transformation".

I think it might be worth it to include a `direction` field in the `Transformation` struct, which could hold `PJ_FWD` or `PJ_INV`.  This, with a small change to the compose function, should be lightweight enough but also make `Transformation` a lot more flexible.  With a similar change implemented locally, I can construct e.g. 
igh
whereas with the normal interface I cannot, since I need the inverse transformation for tick placement.

**Comment by visr** [comment]
Interesting. Do you have an example of a transformation for which that is the case, for testing? Probably indeed the FWD and INV method is better, since it seems closer to how PROJ is meant to be used.

**Comment by asinghvi17** [comment]
https://proj.org/operations/projections/igh.html and https://proj.org/operations/projections/nicol.html were what I tested with

**Comment by asinghvi17** [comment]
I've pushed a prototype to the branch `as/transform_direction`.  Unfortunately, we can't simply share the PJ pointer between the original and inverted transformations, because it is unknown which of them will be finalized first.  Instead, I used a similar approach of extracting the CRS as PJs and then creating a new Transformation with them.

**Comment by c42f** [comment]
Nice! Congratulations on merging this :partying_face: :tada:

**Pull request merged**